include/umf/memory_provider.h

```diff
@@ -21,7 +21,8 @@ extern "C" {
 typedef enum umf_memory_visibility_t {
     UMF_MEM_MAP_PRIVATE = 1, ///< private memory mapping
     UMF_MEM_MAP_SHARED,      ///< shared memory mapping (Linux only)
-    UMF_MEM_MAP_SYNC, ///< direct mapping of persistent memory (supported only for files supporting DAX, Linux only)
+    UMF_MEM_MAP_SYNC =
```

Comment:
i think __attribute__((deprecated)) should work here

Reply:
please just redefine new flag (remove "UMF_MEM_MAP_SYNC =")

Reply:
On the UMF tech meeting, we agreed that `UMF_MEM_MAP_SYNC` can be simply removed.

It is an API/ABI breaking change but in this particular case, it is OK because we are mostly sure that nobody uses it.

Reply:
Removed.
Done